python/ray/train/tests/test_state.py

```diff
@@ -0,0 +1,233 @@
+        }
+    ],
+    "datasets": [
+        {
+        "name": "train",
+        "dataset_name": "train_dataset",
+        "dataset_uuid": "1"
+        }
+    ]
+}"""
+
+
+def _get_run_info_sample(run_id=None, run_name=None) -> TrainRunInfo:
+    dataset_info = TrainDatasetInfo(
+        name="train", dataset_name="train_dataset", dataset_uuid="1"
+    )
+
+    worker_info_0 = TrainWorkerInfo(
+        actor_id="3d86c25634a71832dac32c8802000000",
+        world_rank=0,
+        local_rank=0,
+        node_rank=0,
+        node_id="b1e6cbed8533ae2def4e7e7ced9d19858ceb1ed8ab9ba81ab9c07825",
+        node_ip="10.0.208.100",
+        pid=76071,
+        gpu_ids=[0],
+    )
+
+    worker_info_1 = TrainWorkerInfo(
+        actor_id="8f162dd8365346d1b5c98ebd7338c4f9",
+        world_rank=1,
+        local_rank=1,
+        node_rank=0,
+        node_id="b1e6cbed8533ae2def4e7e7ced9d19858ceb1ed8ab9ba81ab9c07825",
+        node_ip="10.0.208.100",
+        pid=76072,
+        gpu_ids=[1],
+    )
+
+    run_info = TrainRunInfo(
+        name=run_name if run_name else "default_run",
+        id=run_id if run_id else "ad5256bc64c04c83833a8b006f531799",
+        job_id="0000000001",
+        controller_actor_id="3abd1972a19148d78acc78dd9414736e",
+        workers=[worker_info_0, worker_info_1],
+        datasets=[dataset_info],
+    )
+    return run_info
+
+
+def test_schema_equivalance():
+    json_sample = RUN_INFO_JSON_SAMPLE
+    dict_sample = json.loads(RUN_INFO_JSON_SAMPLE)
+
+    run_info_from_json = TrainRunInfo.parse_raw(json_sample)
+    run_info_from_obj = TrainRunInfo.parse_obj(dict_sample)
+
+    # Test serialization equivalence
+    assert run_info_from_json == run_info_from_obj
+
+    # Test dict deserialization equivalence
+    assert run_info_from_json.dict() == dict_sample
+
+    # Test json deserialization equivalence
+    assert json.loads(run_info_from_json.json()) == json.loads(json_sample)
+
+    # Test constructors equivalence
+    assert _get_run_info_sample() == run_info_from_json
+
+
+def test_state_actor_api():
+    state_actor = get_or_create_state_actor()
+    named_actors = ray.util.list_named_actors(all_namespaces=True)
+    assert {
+        "name": TRAIN_STATE_ACTOR_NAME,
+        "namespace": TRAIN_STATE_ACTOR_NAMESPACE,
+    } in named_actors
+
+    # Concurrently register 100 runs
+    num_runs = 100
+    info_list = [_get_run_info_sample(run_id=str(i)) for i in range(num_runs)]
+    ray.get([state_actor.register_train_run.remote(run) for run in info_list])
+
+    # Test get all runs
+    train_runs = ray.get(state_actor.get_all_train_runs.remote())
+    assert len(train_runs) == num_runs
+
+    # Test get a single run by run_id
+    for i in range(num_runs):
+        run_info = ray.get(state_actor.get_train_run.remote(run_id=str(i)))
+        assert run_info == info_list[i]
+
+
+@pytest.mark.parametrize("gpus_per_worker", [0, 1, 2])
+def test_track_e2e_training(ray_start_gpu_cluster, gpus_per_worker):
```

Comment:
Maybe we could test `TrainStateManager` only instead of `DataParallelTrainer`?

Reply:
I'll add another test with `TrainStateManager` only. 

After we put `worker_group` as an argument of `register_train_run`, we will no longer need to test it together with `DataParallelTrainer`.

Reply:
Added a unit test directly with `TrainStateManager`.